src/yunohost/tools.py

```diff
@@ -982,208 +1019,206 @@ def tools_migrations_list(pending=False, done=False):
     """
     Perform migrations
+
+    targets        A list migrations to run (all pendings by default)
+    --skip         Skip specified migrations (to be used only if you know what you are doing) (must explicit which migrations)
+    --auto         Automatic mode, won't run manual migrations (to be used only if you know what you are doing) (must explicit which migrations)
+    --force-rerun  Re-run already-ran migrations (to be used only if you know what you are doing)(must explicit which migrations)
+    --revert       Attempt to revert already-ran migrations (must explicit which migrations)
+    --accept-disclaimer  Accept disclaimers of migrations (please read them before using this option) (only valid for one migration)
     """
 
-    # state is a datastructure that represents the last run migration
-    # it has this form:
-    # {
-    #     "last_run_migration": {
-    #             "number": "00xx",
-    #             "name": "some name",
-    #         }
-    # }
-    state = tools_migrations_state()
+    all_migrations = _get_migrations_list()
 
-    last_run_migration_number = state["last_run_migration"]["number"] if state["last_run_migration"] else 0
+    # Small utility that allows up to get a migration given a name, id or number later
+    def get_matching_migration(target):
+        for m in all_migrations:
+            if m.id == target or m.name == target or m.id.split("_")[0] == target:
+                return m
 
-    # load all migrations
-    migrations = _get_migrations_list()
-    migrations = sorted(migrations, key=lambda x: x.number)
+        raise YunohostError("migrations_no_such_migration", id=target)
 
-    if not migrations:
-        logger.info(m18n.n('migrations_no_migrations_to_run'))
-        return
-
-    all_migration_numbers = [x.number for x in migrations]
 
-    if target is None:
-        target = migrations[-1].number
+    # auto, skip, revert and force are exclusive options
+    if auto + skip + revert + force_rerun > 1:
+        raise YunohostError("migrations_exclusive_options")
 
-    # validate input, target must be "0" or a valid number
-    elif target != 0 and target not in all_migration_numbers:
-        raise YunohostError('migrations_bad_value_for_target', ", ".join(map(str, all_migration_numbers)))
+    # If no target specified
+    if not targets:
+        # skip, revert or force require explicit targets
+        if (revert or force_rerun):
+            raise YunohostError("migrations_must_provide_explicit_targets")
 
-    logger.debug(m18n.n('migrations_current_target', target))
+        # Otherwise, targets are all pending migrations
+        targets = [m for m in all_migrations if m.state == "pending"]
 
-    # no new migrations to run
-    if target == last_run_migration_number:
+    # If explicit targets are provided, we shall validate them
+    else:
+        targets = [get_matching_migration(t) for t in targets]
+        done = [t.id for t in targets if t.state != "pending"]
```

Comment:
```suggestion
        done = [t.id for t in targets if t.state != "done"]
```

?

Reply:
Current code looks right, to have "done" target you just want exclude the "pending" ones (other states are "done" too)

Reply:
done and skip but i think we want to see done and skip if you put --done

Reply:
I think webadmin use --done, and we want to display skipped migrations as done

Reply:
Marking as resolved (c.f. Bram's comment)